### ISSUE TITLE
VSCode won't highlight definitions when running on WSL and in some cases will highlight the .c file partly.

### ISSUE THREAD

**Issue by poltamtamis** [microsoft/vscode-cpptools#5562]
- OS and Version: Windows 10 1909
- VS Code Version: C/CPP extension
- C/C++ Extension Version: ms-vscode.cpptools v 0.28.1

VSCode seems to be having issues highlighting definitions  when running in WSL and sometimes, in all other modes, highlights the definitions only in a few instances in the .c file. 
What works (almost) properly:
-VSCode with WSL and .c file on C++ language mode
or -VSCode without WSL and pretty much anything
What doesn't work properly:
-VSCode with WSL and .c file on C language mode (screenshots below that explain this)

**Steps to reproduce**
Opening this with WSL and C Lang mode is enough:
```
#ifndef GRAPH_H
#define GRAPH_H
#define MAXVERTEX 20

typedef enum {FALSE, TRUE} Boolean;

typedef int Vertex;

typedef struct edge {
   Vertex endpoint;
   struct edge *nextedge;
} Edge;

typedef struct graph {
   int n;     /* number of vertices in the graph */
   Edge *firstedge[MAXVERTEX];
} Graph;
Graph Initialize(Graph);
Edge CreateEdge(Vertex,Vertex,Edge*,Edge*);
Graph InsertEdge(Graph,Edge);
void ShowGraph(Graph);
void DepthFirst(Graph);
void traverse(Graph,int);
```
**Expected behavior**
Everything should be highlighted.

**Screenshots**
Without WSL OR with WSL+ C++ lang mode (in both cases 1-2 definitions aren't highlighted):
![image](https://user-images.githubusercontent.com/59566389/82844481-5cd12380-9ee9-11ea-84e2-70c09b7594e1.png)
For the part 'some definitions aren't highlighted' this is an example:
![image](https://user-images.githubusercontent.com/59566389/82844503-6e1a3000-9ee9-11ea-9355-404bc15d1ab0.png) (Edge* is highlighted in some parts and in others, not)

With WSL+ C language mode, this is how it looks:
![image](https://user-images.githubusercontent.com/59566389/82844533-7ffbd300-9ee9-11ea-9d6e-b8c5349a8eba.png)

**Comment by Colengms** [comment]
Hi @polt13 .  There are 2 types of colorization in C/C++ files, currently.  The first, syntactic pass is done by VS Code and is based on Text Mate grammar.  (That grammer is contributed from [this](https://github.com/jeff-hykin/cpp-textmate-grammar) repo).  The second pass is semantic, and contributed by us.  Our implementation of semantic colorization was prior to the addition of a Semantic Tokens API in VS Code, and we're in the process of switching to it.  In the meanwhile, our approach was not supportable via remote.  That is tracked by: https://github.com/microsoft/vscode-cpptools/issues/3942

If the colorization issue you are seeing is still apparent with `"C_Cpp.enhancedColorization": "Disabled"`, you might consider trying the "Better C++ Syntax" extension, which is built from https://github.com/jeff-hykin/cpp-textmate-grammar (And would likely include more recent fixes than present in VS Code), or open a bug against that repo.

**Comment by poltamtamis** [comment]
Hi @Colengms . I tried with the cpptools extension ON and Better C++ Syntax on, with enhanced colorization on and off but the outcome is the same. I also tried the same with cpptools. Not sure if the issue is on my end, because I definitely can't get certain parts to actually color and especially on C language mode it's a lot more problematic. I'm referring to remote, as I said and not locally through Windows (that mode seems to work fine).

Regarding the switch to the new Semantics API, is there an ETA for the switch? Other than that, I am not sure if there is anything else I can try to fix the highlighting.

Thanks!
